### ISSUE TITLE
[功能请求] 增强跨平台支持与大型模型接入方式多样化

### ISSUE THREAD

**Issue by chaofanat** [BAAI-Agents/Cradle#58]
尊敬的Cradle开发团队，

希望我的信息能在您忙碌的日子里带来一些启发。我对cradle非常感兴趣，但我想进行测试时，我观察到以下两个主要改进点：

跨平台支持：
目前，Cradle似乎主要支持macos系统，因为requirements的一些包在windows下无法正确安装，您也注明了所需平台。尽管macos平台覆盖了大部分用户，但若能进一步拓展至**windows**，将极大地提升其通用性和吸引更广泛的用户群体。这可能涉及代码优化以兼容新系统，或是提供详细的文档指导如何在非标准平台上设置Cradle。
大语言模型接入方式多样化：
随着AI技术的飞速发展，特别是大型语言模型的兴起，Cradle中灵活的模型接入方法显得尤为重要。用户可能偏好不同的模型接入方式，比如API调用、本地部署或云服务解决方案。我了解到目前的接入方式有openAI api ，Azure以及Claude api,但对于中国用户，这些接人方式并不友好，如果使用langchain等架构，**增加Qwen或者智谱AI等大模型的接入配置**应当相对容易，而这将足够满足中国用户的使用。

如果这两个问题已有解决方案，请不吝赐教，非常感谢！
 
 最后，希望我的建议能为Cradle带来更好的改变，期待此项目的发展。也感谢Cradle开发团队对AI发展做出的贡献。

**Comment by WeihaoTan** [comment]
Thanks for your insightful suggestions and interest in our project.

For the OS platform, all the environments can be run on Windows and some of them can be run on MacOS, mainly due to some games, like RDR2 and Dealer's Life 2 do not have the MacOS version. Windows is our main OS platform for development and evaluation. Please let us know if you encounter any problems when installing the env on both Windows and MacOS.

For the adaption, currently, even GPT-4o can barely complete the tasks and Claude 3 Opus cannot complete most of the tasks. It is difficult for the current open-source models to finish the tasks, so we do not provide the accessibility of other models in this version. While developing, we do not apply the out-of-box framework, like LangChain for best flexibility, maybe we could use part of their code for the encapsulation of the models' request. We will conduct a study later.

**Comment by luyifanlu** [comment]
> Thanks for your insightful suggestions and interest in our project.
> 
> For the OS platform, all the environments can be run on Windows and some of them can be run on MacOS, mainly due to some games, like RDR2 and Dealer's Life 2 do not have the MacOS version. Windows is our main OS platform for development and evaluation. Please let us know if you encounter any problems when installing the env on both Windows and MacOS.
> 
> For the adaption, currently, even GPT-4o can barely complete the tasks and Claude 3 Opus cannot complete most of the tasks. It is difficult for the current open-source models to finish the tasks, so we do not provide the accessibility of other models in this version. While developing, we do not apply the out-of-box framework, like LangChain for best flexibility, maybe we could use part of their code for the encapsulation of the models' request. We will conduct a study later.

开源模型可能在高难度任务达不到要求，但是可以提供框架，完成一些低要求部分。同时，这个框架也可以作为一个环境模拟器，个人觉得环境模拟器，才是这个项目的意义

**Comment by sabibi12** [comment]
![image](https://github.com/user-attachments/assets/dc61e655-3695-4330-88d8-ff9cb0bc4d1c)
![image](https://github.com/user-attachments/assets/9a35f637-2c44-4e8b-96bb-199448b1b33f)
![image](https://github.com/user-attachments/assets/cff0552e-da2b-4815-8ec8-ff929471e97b)
![image](https://github.com/user-attachments/assets/e5f8151d-e90f-4b7a-a74f-08991bd615d3)

我自制了一套适用于windows系统的rpa api，以上为一个基本的实践，我之前也做过类似LLM搭配RPA的项目，希望能够与贵项目进行合作

**Comment by sabibi12** [comment]
![image](https://github.com/user-attachments/assets/dc61e655-3695-4330-88d8-ff9cb0bc4d1c)
![image](https://github.com/user-attachments/assets/9a35f637-2c44-4e8b-96bb-199448b1b33f)
![image](https://github.com/user-attachments/assets/f1ae43be-36b7-42e8-99f0-a61da95ab8a8)
![image](https://github.com/user-attachments/assets/e5f8151d-e90f-4b7a-a74f-08991bd615d3)
我自制了一套适用于windows系统的rpa api，以上为一个基本的实践，我之前也做过类似LLM搭配RPA的项目，希望能够与贵项目进行合作

**Comment by sabibi12** [comment]
https://github.com/user-attachments/assets/140e0f5d-078f-48fc-8ec4-3d1f9cf91d7c

这是一个demo视频

**Comment by sabibi12** [comment]
https://github.com/user-attachments/assets/70fb754f-fb09-4324-9adf-07c38b4bb0bd

这是一个演示demo视频

**Comment by sabibi12** [comment]
https://github.com/user-attachments/assets/165a33a8-2008-4d34-9af5-2cf1fb19a092


这是一个演示demo视频

**Comment by HaishengLiang** [comment]
I hope can use OpenRouter or some other software like it to make us to use some different LLM  （eg Qianwen2、GLM ....)

**Comment by KJinze** [comment]
Agreed, if there is no built-in model, standard model integration is necessary
